sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -303,28 +303,36 @@ abstract class GpuBroadcastExchangeExecBase(
             // Setup a job group here so later it may get cancelled by groupId if necessary.
             sparkContext.setJobGroup(_runId.toString, s"broadcast exchange (runId ${_runId})",
               interruptOnCancel = true)
-            val batch = withResource(new NvtxWithMetrics("broadcast collect", NvtxColor.GREEN,
-              collectTime)) { _ =>
-              val data = child.executeColumnar().map(cb => try {
-                new SerializeBatchDeserializeHostBuffer(cb)
-              } finally {
-                cb.close()
-              })
-              val d = data.collect()
-              new SerializeConcatHostBuffersDeserializeBatch(d, output)
-            }
-
-            val numRows = batch.numRows
-            checkRowLimit(numRows)
-            numOutputBatches += 1
-            numOutputRows += numRows
+            var dataSize = 0L
+            val broadcastResult =
+              withResource(new NvtxWithMetrics("broadcast collect", NvtxColor.GREEN,
+                collectTime)) { _ =>
+                val childRdd = child.executeColumnar()
+                val data = childRdd.map(cb => try {
```

Comment:
Nit: since we're in the neighborhood, seems like a good candidate for `withResource`.

Reply:
So I tried doing this, but I got a serialization error. It's a little cryptic on what is going on. I tried both `withResource` with a batch specifically (i.e. `withResource(cb)` or with the placeholder `withResource(_)` thinking the placeholder was doing something odd type-wise):

```
E                   Serialization stack:
E                   	- object not serializable (class: org.apache.spark.sql.rapids.execution.GpuBroadcastExchangeExecBase$$anon$1, value: org.apache.spark.sql.rapids.execution.GpuBroadcastExchangeExecBase$$anon$1@7b0c20b2)
E                   	- element of array (index: 0)
E                   	- array (class [Ljava.lang.Object;, size 1)
E                   	- field (class: java.lang.invoke.SerializedLambda, name: capturedArgs, type: class [Ljava.lang.Object;)
```